base/util/url.go

```diff
@@ -24,9 +24,21 @@ import (
 	"github.com/vincent-petithory/dataurl"
 )
 
-func MakeDataURL(contents []byte, currentCompression *string, allowCompression bool) (uri string, gzipped bool, err error) {
+func MakeDataURL(contents []byte, currentCompression *string, allowCompression bool) (uri string, compression *string, err error) {
 	// try three different encodings, and select the smallest one
 
+	// If the config has a non-empty compression field, then the
+	// contents were compressed by the user; we can pick a data URL
+	// encoding but we can't compress again.  In this case we set our
+	// compression return value to null so the caller knows not to
+	// record a translation from input contents to output compression.
+	// In all other cases, we need to explicitly set the compression
+	// field to avoid a child config inheriting an incorrect compression
+	// setting from the parent.
+	if util.NilOrEmpty(currentCompression) {
+		compression = util.StrToPtr("")
+	}
```

Comment:
I had to read this comment and code block twice.  The comment starts out talking about a condition, then talks about the "all other cases" - and the code implements the "all other cases".  I kept expecting it to implement what the *start* of the comment was describing.

Maybe:

```
If the config has an empty/nil compression field, we need to 
explicitly set the compression field to avoid a child config
inheriting an incorrect compression setting from the parent.

Otherwise, If the config has a non-empty compression field, then the
contents were compressed by the user; we can pick a data URL
encoding but we can't compress again.  In this case we set our
compression return value to null so the caller knows not to
record a translation from input contents to output compression.
```

maybe?